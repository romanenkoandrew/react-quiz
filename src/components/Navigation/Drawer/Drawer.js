import React from "react";
import styles from "./Drawer.module.css";
import Backdrop from "../../UI/Backdrop/Backdrop";
import { NavLink } from "react-router-dom";

class Drawer extends React.Component {
  renderLinks(links) {
    return links.map((link, index) => {
      return (
        <li key={index}>
          <NavLink
            to={link.to}
            exact={link.exact}
            activeClassName={styles.active}
            onClick={this.props.onClose}
          >
            {link.label}
          </NavLink>
        </li>
      );
    });
  }
  render() {
    const cls = [styles.Drawer];
    if (!this.props.isOpen) {
      cls.push(styles.close);
    }
    const links = [{ to: "/", label: "Список", exact: true }];
    if (this.props.isAuthenticated) {
      links.push({ to: "/quiz-creator", label: "Создать тест", exact: false });
      links.push({ to: "/logout", label: "Выйти", exact: false });
    } else {
      links.push({ to: "/auth", label: "Авторизация", exact: false });
    }
    return (
      <React.Fragment>
        <nav className={cls.join(" ")}>
          <ul>{this.renderLinks(links)}</ul>
        </nav>
        {this.props.isOpen && <Backdrop onClick={this.props.onClose} />}
      </React.Fragment>
    );
  }
}

export default Drawer;
