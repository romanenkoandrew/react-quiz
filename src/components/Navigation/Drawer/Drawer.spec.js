import React from "react";
import { shallow } from "enzyme";
import Drawer from "./Drawer";

describe("Drawer component", () => {
  it("should render Drawer without props", () => {
    const component = shallow(<Drawer />);
    expect(component).toMatchSnapshot();
  });
  it("should render Drawer with props", () => {
    const component = shallow(<Drawer isOpen isAuthenticated />);
    expect(component).toMatchSnapshot();
  });
});
