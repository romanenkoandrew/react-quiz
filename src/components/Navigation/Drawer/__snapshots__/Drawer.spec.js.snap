// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer component should render Drawer with props 1`] = `
<Fragment>
  <nav
    className="Drawer"
  >
    <ul>
      <li
        key="0"
      >
        <NavLink
          activeClassName="active"
          exact={true}
          to="/"
        >
          Список
        </NavLink>
      </li>
      <li
        key="1"
      >
        <NavLink
          activeClassName="active"
          exact={false}
          to="/quiz-creator"
        >
          Создать тест
        </NavLink>
      </li>
      <li
        key="2"
      >
        <NavLink
          activeClassName="active"
          exact={false}
          to="/logout"
        >
          Выйти
        </NavLink>
      </li>
    </ul>
  </nav>
  <Backdrop />
</Fragment>
`;

exports[`Drawer component should render Drawer without props 1`] = `
<Fragment>
  <nav
    className="Drawer close"
  >
    <ul>
      <li
        key="0"
      >
        <NavLink
          activeClassName="active"
          exact={true}
          to="/"
        >
          Список
        </NavLink>
      </li>
      <li
        key="1"
      >
        <NavLink
          activeClassName="active"
          exact={false}
          to="/auth"
        >
          Авторизация
        </NavLink>
      </li>
    </ul>
  </nav>
</Fragment>
`;
